src/main/java/edu/cornell/kfs/module/ar/service/impl/CuAREmailServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package edu.cornell.kfs.module.ar.service.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.kuali.kfs.module.ar.ArKeyConstants;
+import org.kuali.kfs.module.ar.businessobject.CustomerAddress;
+import org.kuali.kfs.module.ar.document.ContractsGrantsInvoiceDocument;
+import org.kuali.kfs.module.ar.service.impl.AREmailServiceImpl;
+
+import java.text.MessageFormat;
+
+public class CuAREmailServiceImpl extends AREmailServiceImpl {
+    
+    @Override
+    protected String getSubject(ContractsGrantsInvoiceDocument invoice) {
+        String grantNumber = invoice.getInvoiceGeneralDetail().getAward().getProposal().getGrantNumber();
+        String subject;
+        String message;
+        if (StringUtils.isBlank(grantNumber)) {
+            subject = kualiConfigurationService.getPropertyValueAsString(ArKeyConstants.CGINVOICE_EMAIL_SUBJECT_NO_GRANT_NUMBER);
+            message = MessageFormat.format(subject, invoice.getDocumentNumber(), 
+                    invoice.getInvoiceGeneralDetail().getAward().getProposal().getProposalNumber());
```

Comment:
You can get the proposal number from the InvoiceGeneralDetail directly, so you don't have to walk the object graph this far to get it.

Reply:
This code was pull directly from base with the minimal amount of changes performed. 
Do we really want to make this change and deviate even further with this mod?

Reply:
Base code isn't using proposal number here. Nevertheless, after further discussion with Nancy, I'm fine with leaving it like this so that it's consistent with the grant number derivation.